runtime/crab/src/lib.rs

```diff
@@ -0,0 +1,724 @@
+
+#[cfg(feature = "std")]
+include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+
+mod pallets;
+pub use pallets::*;
+
+mod weights;
+
+pub use sp_consensus_aura::sr25519::AuthorityId as AuraId;
+
+// crates.io
+use smallvec::smallvec;
+// cumulus
+use cumulus_pallet_parachain_system::RelayNumberStrictlyIncreases;
+// darwinia
+use dc_primitives::*;
+// polkadot
+use xcm::latest::prelude::BodyId;
+use xcm_executor::XcmExecutor;
+// substrate
+use frame_support::{
+	dispatch::DispatchClass,
+	weights::{
+		ConstantMultiplier, Weight, WeightToFeeCoefficient, WeightToFeeCoefficients,
+		WeightToFeePolynomial,
+	},
+	PalletId,
+};
+use frame_system::EnsureRoot;
+use sp_core::{crypto::KeyTypeId, OpaqueMetadata, H160, H256, U256};
+#[cfg(any(feature = "std", test))]
+pub use sp_runtime::BuildStorage;
+use sp_runtime::{
+	generic,
+	traits::Block as BlockT,
+	transaction_validity::{TransactionSource, TransactionValidity},
+	ApplyExtrinsicResult,
+};
+pub use sp_runtime::{MultiAddress, Perbill, Permill};
+use sp_std::prelude::*;
+#[cfg(feature = "std")]
+use sp_version::NativeVersion;
+use sp_version::RuntimeVersion;
+
+/// Block type as expected by this runtime.
+pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+
+/// A Block signed with a Justification
+pub type SignedBlock = generic::SignedBlock<Block>;
+
+/// The SignedExtension to the basic transaction logic.
+pub type SignedExtra = (
+	frame_system::CheckNonZeroSender<Runtime>,
+	frame_system::CheckSpecVersion<Runtime>,
+	frame_system::CheckTxVersion<Runtime>,
+	frame_system::CheckGenesis<Runtime>,
+	frame_system::CheckEra<Runtime>,
+	frame_system::CheckNonce<Runtime>,
+	frame_system::CheckWeight<Runtime>,
+	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
+);
+
+/// Unchecked extrinsic type as expected by this runtime.
+pub type UncheckedExtrinsic =
+	fp_self_contained::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
+
+/// Extrinsic type that has already been checked.
+pub type CheckedExtrinsic =
+	fp_self_contained::CheckedExtrinsic<AccountId, RuntimeCall, SignedExtra, H160>;
+
+/// Executive: handles dispatch to the various modules.
+pub type Executive = frame_executive::Executive<
+	Runtime,
+	Block,
+	frame_system::ChainContext<Runtime>,
+	Runtime,
+	AllPalletsWithSystem,
+>;
+
+// TODO: move to impl.rs
+pub struct DealWithFees<R>(sp_std::marker::PhantomData<R>);
+impl<R> frame_support::traits::OnUnbalanced<pallet_balances::NegativeImbalance<R>>
+	for DealWithFees<R>
+where
+	R: pallet_balances::Config, /* R: pallet_balances::Config + pallet_treasury::Config,
+	                             * pallet_treasury::Pallet<R>:
+	                             * OnUnbalanced<pallet_balances::NegativeImbalance<R>>, */
+{
+	// this seems to be called for substrate-based transactions
+	fn on_unbalanceds<B>(
+		mut fees_then_tips: impl Iterator<Item = pallet_balances::NegativeImbalance<R>>,
+	) {
+		if let Some(fees) = fees_then_tips.next() {
```

Comment:
Will we deal with it? I want to add unused prefix. @AurevoirXavier @AsceticBear

Reply:
Later. In process.